packages/data-union-solidity/contracts/DataUnionSidechain.sol

```diff
@@ -305,6 +315,12 @@ contract DataUnionSidechain is Ownable, IERC20Receiver, IERC677Receiver {
         activeMemberCount -= 1;
         inactiveMemberCount += 1;
         emit MemberParted(member);
+
+        // listeners do NOT get a chance to prevent parting by reverting
+        for (uint i = 0; i < joinPartListeners.length; i++) {
```

Comment:
I don' know how you would fix it, but is it an attack vector, that I can now add a lot of listeners, that fail, and consume a lot of gas, and then it will always(or at least likelyer) run out of gas on this for loop?